src/executor.rs

```diff
@@ -199,3 +133,96 @@ impl Executor {
         Ok(())
     }
 }
+
+async fn task_result(
+    task: &mut Option<JoinHandle<Result<Option<ExitStatus>>>>,
+) -> Result<Option<ExitStatus>> {
+    match task {
+        Some(handle) => handle.await.unwrap(),
+        None => match AlwaysPending.await {},
+    }
+}
+
+/// A future that will never resolve
+struct AlwaysPending;
```

Comment:
https://doc.rust-lang.org/std/future/fn.pending.html already do this

Reply:
Nice, didn't know that!